src/sklearn_evaluation/plot/agglomerative.py

```diff
@@ -0,0 +1,219 @@
+    values are detected.
+
+    Parameters
+    ----------
+
+    X : array-like, shape = [n_samples, n_features]:
+        Data to cluster, where n_samples is the number of samples and
+        n_features is the number of features.
+
+    Returns
+    -------
+
+    eigvalues : array-like, shape = [n_features, ]
+        Array of eigenvalues obtained from singular value decomposition, used to obtain the optimal cluster
+        distance in scree plot.
+
+    """
+
+    ## Check for missingness in array, if there is replace the values of the arrays with the mean
+    if np.any(np.isnan(X)) == True:
+        warn(
+            "Missing values detected in input matrix. Replacing missing values with respective column means."
+        )
+
+        # compute column means, replace the values of each with the mean of the column
+        means = np.mean(X, axis=0)
+        idx = np.where(np.isnan(X))
+        X[idx] = np.take(means, idx[-1])
+
+    else:
+        X = X
+
+    ## Perform singular Value Decomposition to get Eigenvectors
+    X = np.asmatrix(X.T) * np.asmatrix(X)
+    _, s, _ = svd(X)
+    eigvalues = s**2 / np.sum(s**2)
+
+    return eigvalues
+
+
+### 2) Compute the curvature of the curve in order to get maximum distance between points and projected line ###
+
+
+def best_curvature(eigvalues):
+    """
+    Helper function returning point with max curvature. Obtained by finding the point of elbow
+    curve with maximum distance between line formed by first and last points on the curve. Steps:
+    1) we map a vector between our first and last eigenalues, 2) we calculate the distances between our eigenvalues and
+    point vector, 3) select the eigenvalue that maximizes distance between the vector and eigenvalue
+
+    Parameters
+    ----------
+    eigvalues : array-like, shape = [n_features,]
+        Array of eigenvalues obtained from singular value decomposition. Eigenvalues are
+        descending-ordered - array values and respective indices are utilized to compute the point with max distance
+
+
+    Returns
+    -------
+    opt_clusters : int
+        Recommended number of clusters for agglomerative clustering dendrogram. Listed as subtitle in scree plot.
+
+    """
+
+    # set up array with curve values
+    ticks = len(eigvalues)
+    points = np.vstack((range(ticks), eigvalues)).T
+    first = points[0]
+
+    # find vector and normalize it
+    vector = points[-1] - points[0]
+    norm = vector / (np.sqrt(np.sum(vector**2)))
+
+    # return point with shortest distance between vector between first and last point
+    values = points - first
+    scalar = np.sum(values * repmat(norm, ticks, 1), axis=1)
+    vec_parallel = np.outer(scalar, norm)
+    vec_line = values - vec_parallel
+    dist = np.sqrt(np.sum(vec_line**2, axis=1))
+    opt_clusters = np.argmax(dist)
+
+    return opt_clusters
+
+
+### 3 Scree Plot for optimal cluster cut distance (justification for dendrogram creation) ###
+
+
+def scree_plot(X, ax=None):
```

Comment:
this looks similar to the elbow curve (https://sklearn-evaluation.readthedocs.io/en/latest/api/plot.html#sklearn_evaluation.plot.elbow_curve), is the only difference that this is in the PCA space? Let's ensure we document how is this different from the elbow curve so users know when to use which one

Reply:
also, check the CI, one test is failing

Reply:
Hi! I'm trying to fix this issue right now but I'm not sure where exactly in the document should I separate this from the elbow curve. Does it mean that we want a separate section on the page that is different from the elbow_curve section?

Reply:
Working on separating the plot from elbow_curve on the document page. But I'm not sure which file I should edit to make changes on the document page. Please let me know.

Reply:
I think we can just describe the difference with respect to the elbow curve in the docstring. but I think more  importantly, we should know if this is *the same* as the elbow curve, just with a different metric. in such case, we'd want to delete this one so we don't have duplicates.

any hints @hcho1111 ?

Reply:
So for now, I guess I can start by adding the commits with fixes and open a PR. For the docstring, I'm still slightly confused in terms of judging how different this is from the elbow_curve.

Reply:
@WSShawn let's wait for harisson's response since it will affect implementation.
Focus on other issues for now.

Reply:
ok！